editing.py

```diff
@@ -181,36 +181,40 @@ def add_area(self, area, edit_note, auto=False):
 
         data_template = {
             "edit-area.name": area["name"],
+            # Why is disambiguation named "comment" on MB?
             "edit-area.comment": area["disambiguation"],
             "edit-area.type_id": area["type"],
             "edit-area.edit_note": edit_note,
-            "edit-area.iso_3166_1.0": area["iso_3166_1"],
-            "edit-area.iso_3166_2.0": area["iso_3166_2"],
-            "edit-area.iso_3166_3.0": area["iso_3166_3"],
-            "edit-area.url.0.text": area["external_links"]["text"],
-            "edit-area.url.0.link_type_id": area["external_links"]["type"],
+            "edit-area.iso_3166_1": area["iso_3166_1"],
+            "edit-area.iso_3166_2": area["iso_3166_2"],
+            "edit-area.iso_3166_3": area["iso_3166_3"],
+            "edit-area.url": area["url"],
         }
 
         def create_payload(template):
-            # remove null values from template
-            # template = {k: v for k, v in template.items() if v is not None}
-            temp = {}
-            required_fields = ['name']
-            for k, v in template.items():
-                if v is not None:
-                    temp[k] = v
-                else:
-                    if k in required_fields:
-                        raise Exception("Missing required field: " + k)
-            template = temp
-
+            required_fields = ["name"]
             payload = {}
-            for k, v in template.items():
-                if isinstance(v, list):
-                    for i, val in enumerate(v):
-                        payload[k.replace("0", str(i))] = str(val)
+            for k1, v1 in template.items():
```

Comment:
Can we use more explicit names for the variable? I understand that it's supposed to be a generic key and value variable, is that really the best we can do though?